frontend/src/scenes/insights/views/LineGraph/LineGraph.tsx

```diff
@@ -547,7 +547,12 @@ export function LineGraph_({
                         insightNumericId: insight.id || 'new',
                     }}
                 >
-                    <AnnotationsOverlay chart={myLineChart} chartWidth={chartWidth} chartHeight={chartHeight} />
+                    <AnnotationsOverlay
+                        chart={myLineChart}
+                        chartWidth={chartWidth}
+                        chartHeight={chartHeight}
+                        dates={datasets[0]?.days || []}
```

Comment:
The issue with this, if we use `datasets[0]?.days` outright, then all the dates are represented in annotations, but for dense graphs the annotations overlay becomes infamously much wider than the graph, because annotations are positioned the same as _ticks_ – and on a dense graph there's a many-to-one relationship between dates and ticks (so that the graph is readable).
This means we instead need to group those dates based on ticks. Currently on dense graphs some annotations are possibly omitted – the upside however is that there's no sizing weirdness, which seems better for the _average_ user. To ensure a 100% smooth experience I plan to implement the full fix, but it'll be a bit more complex than this.